### ISSUE TITLE
tooltip background is not applied when using JSON Parser

### ISSUE THREAD

**Issue by mskocik** [amcharts/amcharts5#1426]
**Bug description**

When I define custom background `fill` for series `tooltip` it's is not applied when config is parsed. 

```js
// ...
{
  "tooltip": {
     "type": "Tooltip",
     "settings": {
       "background": {
         "type": "PointedRectangle",
         "settings": {
           "fill": am5.color('#fff')
         }
        },
        "getFillFromSprite": false,
        "getLabelFillFromSprite": false,
        "getStrokeFromSprite": !false,
        "labelText": "{valueY}"
     }
  }
}
```

Only manual property update works.

```js
chart.series.getIndex(0).get('tooltip').get('background').set('fill', am5.color('#fff'))
```

Here is [full example](https://jsfiddle.net/h5x70f48/1/).

Or am I doing it wrong?

**Comment by mskocik** [comment]
As I went through source of tooltip, I can see, that `background` property is set in `_afterNew()` method. So not sure it's technically a bug or rather just non-complete API support for JSON parser.

So for this use case (trying to customize something, that is being set _after_) some additional property could be provided, under which similar settings could be set.

Example config (let's use tooltip from above):
```js
{
  tooltip: {
    type: 'Tooltip',
    settings: {
      // direct tooltip settings
    },
    afterNew: {
      background: {
        fill: am5.color('#fff')
      },
      'more.nested.property': {
        someProp: 'value'
      }
    }
  }
}

// som basically we need to execute this
// e.get('background').setAll(config.afterNew.background);
// e.get('more').get('nested').get('property').setAll(config.afterNew['more.nested.property])

// which could be done like this:
const entity = Tooltip.new(root, config.settings);

Object.keys(config.afterNew).forEach(prop => {
  const nested = prop.split('.');
  // nested property getter
  const prop = nested.reduce((el, prop) => el.get(prop), entity);
  prop.setAll(config.afterNew[prop]);
});
```

Probably it's quite simplified, but having something like this would be really awesome.

**Comment by Pauan** [comment]
Thanks for the report, and the jsfiddle is really helpful. We were able to fix the issue, the fix will be in the next version.

**Comment by martynasma** [comment]
Fixed in [5.9.0](https://github.com/amcharts/amcharts5/releases/tag/5.9.0).

## [5.9.0] - 2024-04-09

### Added
- `zoomToDataItems(dataItems, rotate)` method added to `MapPolygonSeries`.
- Bigger hit circle added to Stock charts drawings bullets, which becomes visible when hovered.
- A public method `markDirtyKey()` added. Could be used to trigger an adapter.
- A "Reset to default" link added to `StockChart`'s settings modal.
- New `StockChart`/`DrawingSeries` methods added related to drawing selection: `selectDrawing(id, keepSelection)`, `unselectDrawing(id)`, `unselectDrawings()`.
- New `StockChart`/`DrawingSeries` methods added related to drawing deletion: `deleteDrawing(id)`, `deleteSelectedDrawings()`.
- New `StockChart` events: `drawingadded`, `drawingremoved`, `drawingselected`, `drawingunselected`.
- New `DropdownListControl` methods: `getItemById(id)` and `setItemById(id)`.
- New `IconControl` methods: `getIconByPath(path)` and `setIconByPath(path)`.
- New drawing tools added to `StockChart`: "Triangle" and "Polyfill".

### Changed
- Improved handling of moving lines of an `XYCursor`. It will now move by one cell automatically, or by 5 cells if pressed together with `CTRL`.
- Improved styling of a color picker control in `StockChart`'s toolbar and settings modals.
- Drawing functionality was completely revamped: all drawings are now selectable, editable, movable, and can be deleted by selecting them and pressing `DEL`.
- Accessibility: Pressing SPACE when element with `role="checkbox"` set is focused, will now toggle it just like pressing ENTER.
- `pdfmake` updated to `0.2.10`.

### Fixed
- `Sprite` was not marking its bounds dirty when margins changed. It was causing layouts with margins not to be redrawn after margins changed.
- When drawing mode was enabled in a `StockChart` while in the percent scale, it used to reset scale to regular.
- `Tooltip` no longer overwrites user-provided `background`.
- `Treemap` was not redrawing when layout algorithm was changed dynamically.
- `Tooltip` position was not being updated when bounds of a `Sprite` changed.
- Moving line of an `XYCursor` with a keyboard press could result in an axis line to be out of sync with series tooltips.
- `tabindexOrder` was not working properly in some cases.
- Font family and font size controls were showing unnecessary "+" icon.
- In some rare cases setting value on an object / data item was not working.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
